services/pgnotifier/pgnotifier.go

```diff
@@ -219,6 +235,63 @@ func (notifier *PgNotifierT) trackBatch(batchID string, ch *chan []ResponseT) {
 	})
 }
 
+func (notifier *PgNotifierT) trackAsyncBatch(batchID string, ch *chan []ResponseT) {
+	rruntime.GoForWarehouse(func() {
+		for {
+			time.Sleep(trackBatchInterval)
+			// keep polling db for batch status
+			// or subscribe to triggers
+			pkgLogger.Info("Tracking the async batch that's uploaded")
+			stmt := fmt.Sprintf(`SELECT count(*) FROM %s WHERE batch_id='%s' AND status!='%s' AND status!='%s'`, queueName, batchID, SucceededState, AbortedState)
+			var count int
+			err := notifier.dbHandle.QueryRow(stmt).Scan(&count)
+			if err != nil {
+				pkgLogger.Errorf("PgNotifier: Failed to query for tracking jobs by batch_id: %s, connInfo: %s, error : %s", stmt, notifier.URI, err.Error())
+				continue
+			}
+
+			if count == 0 {
+				stmt = fmt.Sprintf(`SELECT payload, status, error FROM %s WHERE batch_id = '%s'`, queueName, batchID)
+				rows, err := notifier.dbHandle.Query(stmt)
+				if err != nil {
+					panic(err)
+				}
+				var responses []ResponseT
+				for rows.Next() {
+					var status, jobError sql.NullString
+					var payload json.RawMessage
+
+					err = rows.Scan(&payload, &status, &jobError)
+					if err != nil {
+						panic(fmt.Errorf("failed to scan result from query: %s\nwith Error : %w", stmt, err))
```

Comment:
Check with @abhimanyubabbar on this.

Reply:
Also, if warehouse master restarts what will happen.

Reply:
What will happen if slaves got blocked on running some Query? let's discuss with @abhimanyubabbar on this as well. Since we have since for warehouse Queries getting struck for hours as well for. 
Let's have some timeouts on the slave end. If it timeouts it should be a failure scenario.

Reply:
Can we perform there operations again in case of errors or Warehouse master restarts.